p2p/src/channels/rpc/mod.rs

```diff
@@ -108,15 +109,30 @@ impl Default for P2pRpcRequest {
     }
 }
 
+fn to_binary(bytes: &[u8]) -> String {
+    bytes
+        .into_iter()
+        .copied()
+        .map(|byte| {
+            (0..8)
+                .into_iter()
+                .map(|b| byte & (1 << (7 - b)) != 0)
+                .map(|b| if b { '1' } else { '0' })
+                .collect::<String>()
+        })
+        .collect()
+}
```

Comment:
@sebastiencs here I've replicated `Into<Address>::to_string()` behaviour from `ledger` crate. Can you double-check the `Address::to_string()` implementation? There's something fishy about it: we're getting bit by bit, but only `length` times, while when constructing an address we use that `length` to count bytes, not bits...

https://github.com/openmina/openmina/blob/build%2Fgroom-deps/core/src/ledger/address/raw.rs#L482

Reply:
> when constructing an address we use that length to count bytes

Do you mean when we construct an `Address` from a string ? ([here ?](https://github.com/openmina/openmina/blob/8e68037aafddd43842a54c8439baeafee4c6e1eb/ledger/src/address/raw.rs#L20))

That's because the string representation uses 1 byte for each bit:
- `"01011"`: those 5 bytes represent 5 bits.
- `0b01011` Here the bits/native format are just 5 bits

An address of depth 5 needs 5 bits, or 5 bytes in string format

Reply:
No, I was looking at `to_string` and `from()` impls. I've been mislead by `lenght` used in two different contexts in the later one:
```rust
...
        let common_length = NBYTES.min(pos.len());
...
            length: depth.as_u64() as _,
```

The functionality is fine.